site/docs/v1/tech/migration-guide/supabase.adoc

```diff
@@ -0,0 +1,299 @@
+** <<The Final Destination of Imported Users>>
+* <<What to Do Next>>
+* <<Additional Support>>
+
+
+There are a number of different ways applications can be integrated with Supabase, and it would be difficult to cover them all. This guide mentions the typical parts of a bulk migration and focuses on migrating user data from a Supabase-managed user database into FusionAuth.
+
+== Planning Considerations
+
+=== Obtaining User Data
+
+You can write a custom psql script in the Supabase SQL Editor to export user data as JSON that you can import to FusionAuth. In Supabase, user data is stored in the `auth.users` table, and you can join the query with data from other `auth` tables to pull specific attributes.
+
+=== Mapping User Attributes
+
+include::docs/v1/tech/migration-guide/_mapping-user-attributes.adoc[]
+
+
+=== Social Logins
+
+include::docs/v1/tech/migration-guide/_social-login-note.adoc[]
+
+include::docs/v1/tech/migration-guide/_social-login-migration.adoc[]
+
+=== Other Entities
+
+:other_migrated_entities: connections or roles
+include::docs/v1/tech/migration-guide/_other-entities-intro.adoc[]
+:other_migrated_entities!:
+
+* In Supabase, sign-in providers are a source of data for users. FusionAuth calls these Identity Providers.
+* Projects are a high-level construct that groups entities such as users and applications together. FusionAuth calls these Tenants.
+* Supabase has other products your applications might be using, like Edge functions. FusionAuth does not have equivalents for these products, and is focused only on authentication.
+
+==== Identifiers
+
+When creating an object with the FusionAuth API, you can specify the Id. It must be a link:/docs/v1/tech/reference/data-types#uuids[UUID].
+
+This works for users, applications, and tenants, among others.
+
+Supabase uses a custom UUID format. When migrating, a new UUID will be created for the user on FusionAuth, or you can use the Supabase user Id migration.
+
+If you have external dependencies on an Id stored in Supabase, you can add a new attribute under `user.data` (such as `user.data.originalId`) with the value of the Supabase Id. Because everything under `user.data` is indexed and available via search, you'll be able to find your users using either the new FusionAuth Id, or the original Supabase one.
+
+== Exporting Users
+
+To export users from Supabase, you’ll perform the following steps:
+
+1. Access the Supabase SQL Editor and create a new query.
+2. Paste the `ufn_get_user_migration_data_json()` function code into the editor.
+3. Run the function in the editor and save the JSON data to a file, for example, `users.json`.
+
+The script below is a user-defined psql function for exporting user data from Supabase. The script will generate a JSON array object.
+
+[source,shell,title="Supabase user data export function"]
+----
+include::https://raw.githubusercontent.com/ritza-co/fusionauth-import-scripts/supabase/supabase/supabase_user_migration_script.sql[]
```

Comment:
Please open a PR so I can merge this into fusionauth/fusionauth-import-scripts.

Reply:
oops, here it is now https://github.com/FusionAuth/fusionauth-import-scripts/pull/11